tcms_junit_plugin/__init__.py

```diff
@@ -37,13 +39,23 @@ def parse(
                 cases = []
                 for suite in xml:
                     for case in suite:
+                        # Retain the suite name (if present) with each testcase.
+                        if suite.name:
+                            case.suitename = suite.name
                         cases.append(case)
             # or directly <testsuite> (only 1) tag - nose & py.test
             else:
                 cases = list(xml)
 
+            summary_template = Template(self.summary_template)
             for xml_case in cases:
-                summary = f"{xml_case.classname}.{xml_case.name}"[:255]
+                # Only permit non-secret values in this map
+                # Users with template control can retrieve any value set here.
+                values = {
+                    "classname": xml_case.classname, "name": xml_case.name,
+                    "suitename": xml_case.suitename}
+
+                summary = summary_template.substitute(values)[:255]
```

Comment:
I think it's better to move this into a method of the class so that others can override if desired and here just call `self.test_case_summary(values)` or something like that. 

And inside of this method you can default to the previous implementation in case the template isn't provided.

Reply:
Agreed this would be better, but making this plugin extensible in that way properly, probably also wants to pass through both the testsuite (or none if it's that funny type of junit) and testcase to the summary each time. ATM we'd just pass through the test case with the suitename randomly overriden on the python object. 

I don't really want to get involved with unpicking all of that properly; will you accept the PR as the slight improvement it is without this internal refactoring?

Reply:
Fair enough. I was thinking for the very straight forward implementation of passing only the context variables to the method without the rest of the XML data and extend in the future if needed. 

I will accept the PR as-is but will make a few more internal changes to separate this code snippet into its own method before pushing a new release to PyPI. Would delay the release by a few days.